backend/src/apiserver/server/util.go

```diff
@@ -356,6 +356,11 @@ func isAuthorized(resourceManager *resource.ResourceManager, ctx context.Context
 		glog.Infof("Multi-user shared read mode is enabled. Request allowed: %+v", resourceAttributes)
 		return nil
 	}
+	if resourceAttributes.Namespace == "" {
```

Comment:
I do realize that `isAuthorized` is being called by non-pipeline functions and maybe be better to wrap it in another function like in the [upload pipeline](https://github.com/kubeflow/pipelines/blob/0c0e4b61050e6fea6f20dd35e14eb3edc214bbba/backend/src/apiserver/server/pipeline_upload_server.go#L224) to check for empty namespace instead of adding it to this function, looking for feedback

Reply:
@annajung I took a look at other objects that use `isAuthorized` function - in their case, it looks that providing an empty namespace is invalid. For example, see here:
https://github.com/kubeflow/pipelines/blob/0c0e4b61050e6fea6f20dd35e14eb3edc214bbba/backend/src/apiserver/server/job_server.go#L102-L104
https://github.com/kubeflow/pipelines/blob/0c0e4b61050e6fea6f20dd35e14eb3edc214bbba/backend/src/apiserver/server/experiment_server.go#L229-L235

So the auth could be skipped when namespace==' ' only in the case of pipelines.

Reply:
But actually, if they all do their validation earlier, skipping the auth here  shouldn't lead to any problem

Reply:
Right, I did see that it does checks for namespace before calling the `isAuthorized` function, so adding the namespace check here would not affect those functions. 

I like clarification on what "shared pipeline" means in the context of runs and experiments. Does a shared pipeline mean that its runs and experiments are also shared?

Reply:
code updated to remove the namespace check from `isAuthorized` and add it to the createPipeline function instead.